sys/net/application_layer/gcoap/Kconfig

```diff
@@ -0,0 +1,143 @@
+# Copyright (c) 2019 HAW Hamburg
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+menuconfig KCONFIG_MODULE_GCOAP
+    bool "Configure Gcoap"
+    depends on MODULE_GCOAP
+    help
+        Configure Gcoap module using Kconfig. If not set default values and
+        CFLAGS will be used.
+
+if KCONFIG_MODULE_GCOAP
+
+menu "Buffer Sizes"
+
+config GCOAP_PDU_BUF_SIZE
+    int "Request or response buffer size"
+    default 128
+    help
+        Size of the buffer used to build a CoAP request or response.
+
+config GCOAP_REQ_OPTIONS_BUF
+    int "Request options buffer size"
+    default 4
+    help
+        Reduce payload length by this value for a request.
+        Accommodates writing Content-Format option in gcoap_finish(). May be
+        set to zero if the function is not used.
+
+config GCOAP_RESP_OPTIONS_BUF
+    int "Response options buffer size"
+    default 4
+    help
+        Reduce payload length by this value for a response.
+        Accommodates writing Content-Format option in gcoap_finish(). May be
+        set to zero if the function is not used.
+
+config GCOAP_OBS_OPTIONS_BUF
+    int "Observe notification options buffer size"
+    default 4
+    help
+        Reduce payload length by this value for an observe notification.
+        Accommodates writing Content-Format option in gcoap_finish(). May be
+        set to zero if the function is not used.
+
+endmenu # Buffer Sizes
+
+menu "Observe options"
+
+config GCOAP_OBS_CLIENTS_MAX
+    int "Maximum number of Observe clients"
+    default 2
+
+config GCOAP_OBS_REGISTRATIONS_MAX
+    int "Maximum number of registrations for Observable resources"
+    default 2
+
+config GCOAP_OBS_VALUE_WIDTH
```

Comment:
@kb2ma looking at https://github.com/RIOT-OS/RIOT/blob/f6d750afe9420761dd3fa13a5791a35ca4a046f5/sys/include/net/gcoap.h#L581-L587

Is the range of this value [1 - 3]?

Reply:
As it is, the option must take one of these values, so I added the range in ad75697d891699f19a390e3ff83e626a3b0292ef.